packages/core/src/lib/__tests__/modal.spec.ts

```diff
@@ -1,38 +1,61 @@
-import { describe, it, expect, vi } from 'vitest';
-import { render, fireEvent } from '@testing-library/svelte';
+import { describe, it, expect, beforeEach } from 'vitest';
+import { render, fireEvent, screen, within } from '@testing-library/svelte';
+import userEvent from '@testing-library/user-event';
 import Modal from '../modal.svelte';
+import { writable, get } from 'svelte/store';
 
 describe('Modal', () => {
-  it('should emit close event when close icon button is clicked', async () => {
-    const { getByTitle, component } = render(Modal, {
-      open: true,
-    });
+  let isOpen = writable(true);
 
-    const closeButton = getByTitle('Close modal');
-    const onClose = vi.fn();
-    component.$on('close', onClose);
-    await fireEvent.click(closeButton);
+  beforeEach(() => {
+    isOpen = writable(true);
+  });
+
+  it('should close modal when close icon button is clicked', async () => {
+    render(Modal, { isOpen });
+
+    const modal = screen.getByRole('dialog');
+    const closeButton = within(modal).getByRole('button', { name: /close/iu });
 
-    expect(onClose).toBeCalledWith(expect.objectContaining({ detail: true }));
+    await userEvent.click(closeButton);
```

Comment:
Rather than using `userEvent` directly, you should create a user object from `userEvent.setup()`, as [recommended in the docs](https://testing-library.com/docs/user-event/intro/#writing-tests-with-userevent)